addons/web/static/tests/core/select_menu.test.js

```diff
@@ -0,0 +1,1071 @@
+import { expect, test } from "@odoo/hoot";
+import { click, edit, press, queryAllTexts, queryOne } from "@odoo/hoot-dom";
+import { animationFrame, runAllTimers } from "@odoo/hoot-mock";
+import { Component, useState, xml } from "@odoo/owl";
+import { mountWithCleanup, patchWithCleanup } from "@web/../tests/web_test_helpers";
+import { MainComponentsContainer } from "@web/core/main_components_container";
+import { SelectMenu } from "@web/core/select_menu/select_menu";
+
+/**
+ * This utils mounts the Component and the MainComponentContainer
+ * inside the same App (unlike the default `mountWithCleanup`), this
+ * ensures refs and useEffects that target elements inside the menu
+ * still work properly.
+ */
+async function mountSingleApp(ComponentClass, props) {
+    class TestComponent extends Component {
+        static props = { components: { type: Array } };
+        static template = xml`
+            <t t-foreach="props.components" t-as="comp" t-key="comp.component.name">
+                <t t-component="comp.component" t-props="comp.props"/>
+            </t>
+        `;
+        get defaultComponent() {
+            return this.__owl__.bdom.children[0].child.component;
+        }
+    }
+
+    const comp = await mountWithCleanup(TestComponent, {
+        props: {
+            components: [
+                { component: ComponentClass, props: props || {} },
+                { component: MainComponentsContainer, props: {} },
+            ],
+        },
+        noMainContainer: true,
+    });
+
+    return comp.defaultComponent;
+}
+
+class Parent extends Component {
+    static props = ["*"];
+    static components = { SelectMenu };
+    static template = xml`
+        <SelectMenu
+            choices="choices"
+            value="state.value"
+            onSelect.bind="onSelect"
+        />
+    `;
+    setup() {
+        this.state = useState({ value: "world" });
+        this.choices = [
+            { label: "Hello", value: "hello" },
+            { label: "World", value: "world" },
+        ];
+    }
+    onSelect(value) {
+        this.state.value = value;
+    }
+}
+
+async function open() {
+    if (queryOne(".o_select_menu_toggler")) {
```

Comment:
I don't remember why we did this. 
'_o_select_menu_toggler_' button is always present, right?
I think we could simplify or remove `open()`?

`await contains(".o_select_menu_toggler").click() `

Reply:
It might be from when the select_menu sometimes used an actual select, so yeah I'm gonna remove this